### PR TITLE
Allow to remove times from recurring reservation

### LINKED ISSUE #571: Allow user to remove specific times from recurring reservations
If user is doing recurring reservations, she should be able to remove some reservation times before saving the reservations.

remove-reservations-before-confirming

### PR DESCRIPTION
**Pull request by YoYuUm**
Closes #571

I've updated the numberOfOcurrencesLabel english label so it's easier to understand. Should I change the finnish and swedish labels?

### CODE REVIEW

File: app/shared/compact-reservation-list/CompactReservationList.spec.js [hunk 1]
```diff
@@ -66,4 +68,28 @@ describe('shared/compact-reservation-list/CompactReservationList', () => {
...
+      const props = { reservations, removableReservations };
+      const buttons = getWrapper(props).find({ glyph: 'remove-circle' });
+      expect(buttons).to.have.length(2);
+    });
+
+    it('passes onRemoveClick func to onClick prop', () => {
+      const removableReservations = [Reservation.build()];
+      const props = { removableReservations };
+      const button = getWrapper(props).find({ glyph: 'remove-circle' });
+      expect(defaultProps.onRemoveClick.callCount).to.equal(0);
```
Comment (lraivio) [inline app/shared/compact-reservation-list/CompactReservationList.spec.js]: - [ ] Whenever we are testing functions with mocks, let's try to create the mock on the test that is using it. Trouble with mocks in defaultProps is that they can be used or called in multiple tests which can break some tests if we test callCounts and such. 

In this case you can just create the `onRemoveClick` mock in this test and pass it as a prop.

File: app/shared/compact-reservation-list/CompactReservationList.spec.js [hunk 2]
```diff
@@ -18,6 +19,7 @@ describe('shared/compact-reservation-list/CompactReservationList', () => {
     resources: Immutable({
       [resource.id]: resource,
     }),
+    onRemoveClick: simple.mock(),
```
Comment (lraivio) [inline app/shared/compact-reservation-list/CompactReservationList.spec.js]: - [ ] Change this to just plain arrow function so we don't accidentally use the same mock in multiple tests.
```
onRemoveClick: () => null,
```

File: app/i18n/messages/en.json [hunk 3]
```diff
@@ -105,7 +105,7 @@
   "RecurringReservationControls.frequencyNone": "None",
   "RecurringReservationControls.frequencyWeekly": "Weekly",
   "RecurringReservationControls.lastTimeLabel": "Ends on",
-  "RecurringReservationControls.numberOfOccurrencesLabel": "Occurrences",
+  "RecurringReservationControls.numberOfOccurrencesLabel": "Repeatitions",
```
Comment (lraivio) [inline app/i18n/messages/en.json]: - [ ] Should be "Repetitions".

Comment (lraivio) [inline app/i18n/messages/en.json]: The Finnish and Swedish translations should be alright.

File: app/shared/compact-reservation-list/CompactReservationList.js [hunk 4]
```diff
@@ -18,8 +19,16 @@ class CompactReservationList extends Component {
       <li key={reservation.begin}>
         {resourceName ? `${resourceName}: ` : ''}
         <TimeRange begin={reservation.begin} end={reservation.end} />
+        {removable &&
+          <Glyphicon
+            glyph="remove-circle"
+            onClick={() => this.props.onRemoveClick(reservation.begin)}
```
Comment (YoYuUm) [inline app/shared/compact-reservation-list/CompactReservationList.js]: I can't get the reservation from the props, so even if there would be a handleRemoveClick function, i would need to invoke it from the component like {() => this.handleRemoveClick(reservation)}

**Comment by lraivio** [comment]
Nice work :+1:
You can squash the fixups.

**Pull request merged**